oi/CodeGen.cpp

```diff
@@ -293,95 +369,127 @@ void addStandardGetSizeFuncDefs(std::string& code) {
 void getClassSizeFuncDecl(const Class& c, std::string& code) {
   code += "void getSizeType(const " + c.name() + " &t, size_t &returnArg);\n";
 }
+}  // namespace
 
-void getClassSizeFuncDef(const Class& c,
-                         SymbolService& symbols,
-                         bool polymorphicInheritance,
-                         std::string& code) {
-  bool enablePolymorphicInheritance = polymorphicInheritance && c.isDynamic();
+/*
+ * Generates a getSizeType function for the given concrete class.
+ *
+ * Does not worry about polymorphism.
+ */
+void CodeGen::getClassSizeFuncConcrete(std::string_view funcName,
+                                       const Class& c,
+                                       std::string& code) const {
+  code += "void " + std::string{funcName} + "(const " + c.name() +
+          " &t, size_t &returnArg) {\n";
 
-  std::string funcName = "getSizeType";
-  if (enablePolymorphicInheritance) {
-    funcName = "getSizeTypeConcrete";
+  if (thriftIssetMembers_.find(&c) != thriftIssetMembers_.end()) {
+    code += "  using thrift_data = apache::thrift::TStructDataStorage<" +
+            c.fqName() + ">;\n";
   }
 
-  code +=
-      "void " + funcName + "(const " + c.name() + " &t, size_t &returnArg) {\n";
-  for (const auto& member : c.members) {
+  for (size_t i = 0; i < c.members.size(); i++) {
+    const auto& member = c.members[i];
     if (member.name.starts_with(type_graph::AddPadding::MemberPrefix))
       continue;
+
+    if (const auto it = thriftIssetMembers_.find(&c);
+        it != thriftIssetMembers_.end()) {
+      const Member* issetMember = it->second;
```

Comment:
Is there a reason this is done inside the loop instead of before it? It could avoid repeating this map lookup.

Reply:
I suppose it was so that `it` was scoped to only the places where it was valid and used. I'll try and refactor it to keep the scoping as much as possible without the performance penalty

Reply:
I did:
```
const Member* issetMember = nullptr;
if (const auto it = thriftIssetMembers_.find(&c); it != thriftIssetMembers_.end()) {
  issetMember = it->second;
}
```

Above the loop then checked `issetMember != nullptr`. Seemed to work okay.